modules/snooker/src/main/ts/ephox/snooker/operate/Redraw.ts

```diff
@@ -24,22 +24,33 @@ const render = function <T extends DetailNew> (table: SugarElement, grid: RowDat
     (c) => Fun.curry(Insert.after, c)
   );
 
-  const renderSection = function (gridSection: RowDataNew<T>[], sectionName: 'thead' | 'tbody' | 'tfoot') {
-    const section = SelectorFind.child(table, sectionName).getOrThunk(function () {
+  const insertColGroup = Arr.last(SelectorFilter.children(table, 'caption')).fold(
```

Comment:
This is again a lot of duplication with `insertThead`. Can you please refactor this to remove the duplication, as this could just be a function that takes a selector and then inserts appropriately.